frontend/src/features/admin-form/responses/IndividualResponsePage/IndividualResponsePage.tsx

```diff
@@ -132,6 +143,43 @@ export const IndividualResponsePage = (): JSX.Element => {
     return `..?${searchParams}`
   }, [lastNavPage, lastNavSubmissionId])
 
+  const attachmentDownloadUrls = useMemo(() => {
+    const attachmentDownloadUrls = new Map()
+    data?.responses.forEach(({ questionNumber, downloadUrl, answer }) => {
+      if (!questionNumber || !downloadUrl || !answer) return
+      attachmentDownloadUrls.set(questionNumber, {
+        url: downloadUrl,
+        filename: answer,
+      })
+    })
+    return attachmentDownloadUrls
+  }, [data?.responses])
+
+  const handleDownloadMutation = useMutation(
+    async ({
+      attachmentDownloadUrls,
+      secretKey,
+    }: {
+      attachmentDownloadUrls: AttachmentsDownloadMap
+      secretKey: string
+    }) => {
+      const byteArray = await downloadAndDecryptAttachmentsAsZip(
+        attachmentDownloadUrls,
+        secretKey,
+      )
+      if (!byteArray) throw new Error('Invalid file')
+      return FileSaver.saveAs(
+        new Blob([byteArray]),
+        `RefNo ${submissionId}.zip`,
+      )
+    },
+  )
+
+  const handleDownload = useCallback(() => {
+    if (!secretKey) return
+    return handleDownloadMutation.mutate({ attachmentDownloadUrls, secretKey })
```

Comment:
should we also add error and success toasts to be more consistent?

Reply:
I didn't add them because we don't have them for individual attachment downloads either. Can add if we would like to, should we add for the individual attachment download too?